### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by tireddy2**

### CODE REVIEW

File: draft-reddy-add-resolver-info-06.txt [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+      name in the URL and checking if the subjectAltName entry in the
+      server certificate includes the name of the encrypted DNS server.
+      If this match fails, the client MUST ignore the resolver
+      information.  As such, the URL should be treated only as
+      diagnostic information for IT staff.  This is a mandatory
+      attribute.
+
+   New keys can be defined as per the procedure defined in Section 7.2.
+
+   Figure 1 shows an example of resolver information.
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt:204]: I suggest we transform this example into a record as will be published in DNS.

File: draft-reddy-add-resolver-info-06.txt [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+   The client can retrieve the resolver information using the RESINFO
+   RRtype and QNAME of the domain name that is used to authenticate the
+   DNS server (referred to as ADN in [I-D.ietf-add-dnr]).
+
+   If the special use domain name "resolver.arpa" defined in
+   [I-D.ietf-add-ddr] is used to discover the Encrypted DNS server, the
+   client can retrieve the resolver information using the RESINFO RRtype
+   and QNAME of the designated resolver.
+
+4.  Format of the Resolver Information
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt:140]: check if we can sort the formatting requirements into wire and zone formats.

File: draft-reddy-add-resolver-info-06.txt [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+
+   New keys can be defined as per the procedure defined in Section 7.2.
+
+   Figure 1 shows an example of resolver information.
+
+     qnameminimization=true
+     extendeddnserror=15,16,17
+     "resinfourl": "https://resolver.example.com/guide"
+
+                Figure 1: An Example of Resolver Information
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt]: Also, update the name to make use of the keys in the spec

Comment (tireddy2) [reply draft-reddy-add-resolver-info-06.txt]: Yes, fixed.

File: draft-reddy-add-resolver-info-06.txt (hunk 1 above)
Comment (tireddy2) [reply draft-reddy-add-resolver-info-06.txt:204]: Good point, fixed.

File: draft-reddy-add-resolver-info-06.txt (hunk 2 above)
Comment (tireddy2) [reply draft-reddy-add-resolver-info-06.txt:140]: We have the following text to discuss the format rules:

The format rules for TXT records are defined in Section 3.3.14 of the DNS specification [RFC1035] and Section 6.1 of the DNS-based Service Discovery [RFC6763].

File: draft-reddy-add-resolver-info-06.txt [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+      If this match fails, the client MUST ignore the resolver
+      information.  As such, the URL should be treated only as
+      diagnostic information for IT staff.  This is a mandatory
+      attribute.
+
+   New keys can be defined as per the procedure defined in Section 7.2.
+
+   Figure 1 shows an example of resolver information.
+
+  resolver.example.net. 7200  IN RESINFO qnamemin exterr=15,16,17 resinfourl:https://resolver.example.com/guide
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt]:
```suggestion
  resolver.example.net. 7200 IN RESINFO qnamemin exterr=15,16,17 resinfourl=https://resolver.example.com/guide
```

File: draft-reddy-add-resolver-info-06.txt [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+
+             Figure 1: An Example of Resolver Information
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt]:
```suggestion
             Figure 1: An Example of Resolver Information Record
```

File: draft-reddy-add-resolver-info-06.txt [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+      rules for the keys is defined in Section 6.4 of [RFC6763], if
+      there is no '=' in a key, then it is a boolean attribute, simply
+      identified as being present, with no value.  This is a mandatory
+      attribute.
+
+   exterr:  If the DNS server supports extended DNS error (EDE)
+      [RFC8914] to return additional information about the cause of DNS
+      errors, the parameter lists the possible extended DNS error codes
+      that can be returned by the DNS server.  When multiple values are
+      present, they are seperated by comma.  This is an optional
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt]:
```suggestion
      present, these values MUST be comma-separated.  This is an optional
```

File: draft-reddy-add-resolver-info-06.txt [hunk 7]
```diff
@@ -0,0 +1,448 @@
...
+   size are discussed in Section 6.2 of [RFC6763].  The RESINFO RRtype
+   uses key/value pairs to convey the resolver information.  Each key/
+   value pair is encoded using the format rules defined in Section 6.3
+   of DNS-SD [RFC6763].  Using standardized key/value syntax within the
+   RESINFO RRtype makes it easier for additional keys to be defined
+   later.  If an implementation sees unknown keys in a RESINFO RRtype,
+   it MUST silently ignore them.  The rules for the keys is defined in
+   Section 6.4 of [RFC6763].
+
+   All keys MUST either be defined in the IANA registry Section 7.2 or
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt]:
```suggestion
   Keys MUST either be defined in the IANA registry Section 7.2 or
```

File: draft-reddy-add-resolver-info-06.txt [hunk 8]
```diff
@@ -0,0 +1,448 @@
...
+   begin with the substring "temp-" for names defined for local use
+   only.
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt:159]:
```suggestion
   only. Unknown keys are silently discarded by DNS clients.
```

File: draft-reddy-add-resolver-info-06.txt [hunk 9]
```diff
@@ -0,0 +1,448 @@
...
+
+5.  Resolver Information
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt]:
```suggestion
5.  Resolver Information Keys/Values
```

File: draft-reddy-add-resolver-info-06.txt [hunk 10]
```diff
@@ -0,0 +1,448 @@
...
+
+
+
+Reddy & Boucadair         Expires 3 March 2023                  [Page 3]
+

+Internet-Draft          DNS Resolver Information             August 2022
+
+
+   qnamemin:  If the DNS server supports QNAME minimisation [RFC9156] to
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt:173]:
```suggestion
   qnamemin key:  If the DNS server supports QNAME minimisation [RFC9156] to
```

File: draft-reddy-add-resolver-info-06.txt [hunk 11]
```diff
@@ -0,0 +1,448 @@
...
+      improve DNS privacy, the key is present.  Note that, as per the
+      rules for the keys is defined in Section 6.4 of [RFC6763], if
+      there is no '=' in a key, then it is a boolean attribute, simply
+      identified as being present, with no value.  This is a mandatory
+      attribute.
+
+   exterr:  If the DNS server supports extended DNS error (EDE)
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt:180]:
```suggestion
   exterr key:  If the DNS server supports extended DNS error (EDE)
```

File: draft-reddy-add-resolver-info-06.txt [hunk 12]
```diff
@@ -0,0 +1,448 @@
...
+      [RFC8914] to return additional information about the cause of DNS
+      errors, the parameter lists the possible extended DNS error codes
+      that can be returned by the DNS server.  When multiple values are
+      present, they are seperated by comma.  This is an optional
+      attribute.
+
+   infourl:  An URL that points to the generic unstructured resolver
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt:187]:
```suggestion
   infourl key:  An URL that points to the generic unstructured resolver
```

File: draft-reddy-add-resolver-info-06.txt [hunk 13]
```diff
@@ -0,0 +1,448 @@
...
+
+
+8.  Acknowledgments
+
+   This specification leverages the work that has been documented in
+   [I-D.pp-add-resinfo].
+
+   Thanks to Tommy Jensen, Vittorio Bertola, Vinny Parla, Chris Box, Ben
+   Schwartz, Tony Finch, Daniel Kahn Gillmor, Eric Rescorla and Shashank
+   Jain for the discussion and comments.
```
Comment (boucadair) [inline draft-reddy-add-resolver-info-06.txt]:
```suggestion
   Jain for the discussion and comments. Thanks to Mark Andrews, Joe Abley, Paul Wouters, Tim Wicinski,  and Steffen Nurpmeso for the discussion on the RR formatting rules. 
```

File: draft-reddy-add-resolver-info-06.txt (hunk 8 above)
Comment (tireddy2) [reply draft-reddy-add-resolver-info-06.txt:159]: It is already covered in the first paragraph of Section 4 with the following text:
If an DNS client sees unknown keys in a RESINFO RRtype, it MUST silently ignore them.

File: draft-reddy-add-resolver-info-06.txt (hunk 10 above)
Comment (tireddy2) [reply draft-reddy-add-resolver-info-06.txt:173]: I don't see the need to add the keyword "key".

**Pull request merged**